src/platform/Kconfig

```diff
@@ -249,6 +255,7 @@ endchoice
 config MAX_CORE_COUNT
 	int
 	default 5 if LUNARLAKE
+	default 5 if PANTHERLAKE
```

Comment:
It was for compatibility in the past but right now it could be easily removed because we are already using CONFIG_MP_MAX_NUM_CPUS in most places.

In sof the most used core kconfig is CONFIG_CORE_COUNT which is based on zephyr

```
config CORE_COUNT
	int "Number of cores"
	default MP_MAX_NUM_CPUS if KERNEL_BIN_NAME = "zephyr"
	default MAX_CORE_COUNT
```

**so this is a good thing to do but in a separate PR**